iothub/device/samples/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -69,22 +100,55 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
+            string writablePropertyName = "serialNumber";
+            if (properties.Writable.Contains(writablePropertyName))
+            {
+                _logger.LogDebug($"Found writable property request \"{writablePropertyName}\": {properties.Writable[writablePropertyName]}");
+            }
+
+            // see if we have a device reported value for "serialNumber" and Thermostat2."initialValue".
+            if (properties.Contains("serialNumber"))
+            {
+                var serialNumberReported = properties["serialNumber"];
+            }
+
+            if (properties.Contains(Thermostat2) && properties[Thermostat2].ContainsKey("initialValue"))
+            {
+                var initialStateThermostat2 = properties[Thermostat2]["initialValue"];
+            }
+
+            s_applicationStartTime = DateTimeOffset.Now;
+
+            // CommandEventDispatcherAsync is a dispatcher that we provide to dispatch the individual component/ root-level command callbacks.
+            // Alternatively, you can also have an uber callback that implements a dispatcher internally.
+            // The important thing to note here is that you can only set a single callback for subscribing to command events.
+            // If you set multiple callbacks to this API, the latest one will be invoked.
+            _logger.LogDebug("Set handler for 'reboot', 'getMaxMinReport' and 'updateTemperatureWithDelay' commands.");
+            _commandEventCallbacks.Add("reboot", HandleRebootCommandAsync);
+            _commandEventCallbacks.Add("getMaxMinReport", HandleMaxMinReportCommandAsync);
+            _commandEventCallbacks.Add("updateTemperatureWithDelay", HandleTemperatureUpdateCommandAsync);
+            await _deviceClient.SubscribeToCommandsAsync(CommandEventDispatcherAsync, null, s_commandConvention, cancellationToken);
+
+            // WritablePropertyEventDispatcherAsync is a dispatcher that we provide to dispatch the individual component/ root-level property callbacks.
+            // Alternatively, you can also have an uber callback that implements a dispatcher internally.
+            // The important thing to note here is that you can only set a single callback for subscribing to property updates.
+            // If you set multiple callbacks to this API, the latest one will be invoked.
+            _logger.LogDebug("Set handler to receive 'targetTemperature' and 'temperatureRange' updates.");
+            _writablePropertyEventCallbacks.Add(Thermostat1, TargetTemperatureUpdateCallbackAsync);
+            _writablePropertyEventCallbacks.Add(Thermostat2, TargetTemperatureUpdateCallbackAsync);
+            _writablePropertyEventCallbacks.Add("temperatureRange", SendTemperatureRangeAsync);
+            await _deviceClient.SubscribeToWritablePropertyEventAsync(WritablePropertyEventDispatcherAsync, userContext: cancellationToken, cancellationToken: cancellationToken);
```

Comment:
`userContent: cancellationToken` ??

Reply:
yeah, I wanted the cancellation token being observed by the application to be passed to the callback